util/encoding/key_encoding_test.go

```diff
@@ -291,13 +294,76 @@ func disabledTestFloatMandE(t *testing.T) {
 		{1234.5, 2, []byte{0x19, 0x45, 0x64}},
 		{12.345, 1, []byte{0x19, 0x45, 0x64}},
 		{0.123, 0, []byte{0x19, 0x3c}},
+		{-0.123, 0, []byte{0x19, 0x3c}},
 		{0.0123, 0, []byte{0x03, 0x2e}},
 		{0.00123, -1, []byte{0x19, 0x3c}},
-		{9223372036854775807, 10, []byte{0x13, 0x2d, 0x43, 0x91, 0x07, 0x89, 0x6d, 0x9b, 0x75, 0x0e}},
+		// The following value cannot be precisely represented as a float.
+		// {9223372036854775807, 10, []byte{0x13, 0x2d, 0x43, 0x91, 0x07, 0x89, 0x6d, 0x9b, 0x75, 0x0e}},
```

Comment:
Include tests for values outside the range of an int64 and especially those with 3-digit exponents; large numbers can be represented precisely in float64s as long as their low bits are zero.